docs/Minimal_Build__Advanced_Usage_with_NNAPI.md

```diff
@@ -1,77 +1,121 @@
+The 'extended' and 'layout' optimizations replace one or more standard ONNX operators with custom internal ONNX Runtime operators designed to boost performance on certain hardware (HW) platforms. These optimizations are applied to the nodes assinged to the corresponding EP for the target HW platform. These custom internal operators are generally limited to the EPs for execution on x86_64 / aarch64 CPU architecture and the CUDA EP on NVIDIA GPUs.
+
+These optimizations will not be applicable to the nodes assigned to the NNAPI EP.
+
+### Example of optimizations on MNIST
 
 Below is an example of the changes that occur in 'basic' and 'extended' optimizations when applied to the MNIST model.
- - At the 'basic' level we combine the Conv and Add nodes (the addition is done via the 'B' input to Conv), we combine the MatMul and Add into a single Gemm node (the addition is done via the 'C' input to Gemm), and constant fold to remove one of the Reshape nodes. 
- - At the 'extended' level we additionally fuse the Conv and Relu nodes using the internal ONNX Runtime FusedConv operator.
+
+- At the 'basic' level we combine the Conv and Add nodes (the addition is done via the 'B' input to Conv), we combine the MatMul and Add into a single Gemm node (the addition is done via the 'C' input to Gemm), and constant fold to remove one of the Reshape nodes. 
+- At the 'extended' level we additionally fuse the Conv and Relu nodes using the internal ONNX Runtime FusedConv operator.
 
 <img align="center" src="images/mnist_optimization.png" alt="Changes to nodes from basic and extended optimizations."/>
 
-At runtime, for each group of nodes that is assigned to it, the NNAPI EP will create an NNAPI model that replicates the processing done by the nodes in the group. It will create a function to handle executing this NNAPI model, including copying data between the CPU and NNAPI. The original nodes in the model are replaced with a single node containing this function.
+During runtime, for each group of nodes that is assigned to it, the NNAPI EP will create an NNAPI model that replicates the processing done by the nodes in the group. It will create a function to handle executing this NNAPI model, including copying data between the CPU and NNAPI. The original nodes in the model are replaced with a single node containing this function.
 
 ### Considerations when creating an NNAPI-aware ORT format model
 
-There are no optimizations available at runtime for an ORT format model, so we generally want to optimize to the highest level possible when creating the model. That usually means optimizing to the 'extended' level, as 'layout' level optimizations are not recommended (they are device specific). 
+The ONNX model intended for using with NNAPI should be optimized when creating the ORT formatted model. *Step 1* and *Step 2* optimizations should be used.
```

Comment:
If you're creating and NNAPI-aware model you want all the steps to be enabled because you want the extended optimizations to apply

Reply:
Added clarification about Step 3 being model specific.